network/framework/src/protocols/health_checker/mod.rs

```diff
@@ -143,6 +155,13 @@ impl<NetworkClient: NetworkClientInterface<HealthCheckerMsg> + Unpin> HealthChec
         let ticker = self.time_service.interval(self.ping_interval);
         tokio::pin!(ticker);
 
+        let connection_events = self
+            .connection_events_injection
+            .take()
+            .unwrap_or_else(|| self.network_interface.get_peers_and_metadata().subscribe());
```

Comment:
I guess you're doing this only for testing purposes, right? If so, this could be cleaner by just passing in `connection_events` to `start()` directly?

Reply:
I got halfway through coding this just now but because of the builder patterns involved in the unit test code it's much easier to set a value before `start()`

Reply:
It still feels a little weird to see this construction. Passing in the events to the `start()` function will mean you don't need to mix regular and test-only code.

Reply:
Yeah, it's weird, but having done another look at doing it the other way, I still think the other way is messier.